app/server/models/transfer_account.py

```diff
@@ -194,19 +194,38 @@ def get_approval(self, receiving_address):
                 return approval
         return None
 
-    def approve_and_disburse(self):
-        if not self.is_approved:
-            self.is_approved = True
+    def approve_and_disburse(self, initial_disbursement=None, auto_resolve=False):
+        from server.utils.access_control import AccessControl
 
-            if self.is_beneficiary:
-                disbursement = self.make_initial_disbursement()
+        auto_approve = current_app.config['AUTO_APPROVE_TRANSFER_ACCOUNTS']
+        admin = getattr(g, 'user', None)
+
+        if not self.is_approved and admin:
+            if AccessControl.has_sufficient_tier(admin.roles, 'ADMIN', 'superadmin'):
+                auto_resolve = True
+                self.is_approved = True
+
+            elif AccessControl.has_sufficient_tier(admin.roles, 'ADMIN', 'admin'):
+                auto_resolve = initial_disbursement \
+                               <= current_app.config['MAXIMUM_CUSTOM_INITIAL_DISBURSEMENT']
+                self.is_approved = auto_approve
+
+            elif AccessControl.has_sufficient_tier(admin.roles, 'ADMIN', 'subadmin'):
+                auto_resolve = False
+                self.is_approved = False
+
+        if self.is_beneficiary:
+            if len(self.credit_receives) < 1:
+                # approve_and_disburse might be called for a second time to disburse
+                # so first check that no credit transfer have already been received
+                disbursement = self._make_initial_disbursement(initial_disbursement, auto_resolve)
                 return disbursement
 
-    def make_initial_disbursement(self, initial_balance=None):
+    def _make_initial_disbursement(self, initial_disbursement, auto_resolve=False):
```

Comment:
If it's possible for admins to have `current_app.config['AUTO_APPROVE_TRANSFER_ACCOUNTS'] = False`, 
we might consider that initial disbursements only resolve once the user is approved. Otherwise what could happen is:
1. admin creates a user in unapproved state
2. disbursement goes through automatically because it's pending
3. superadmin elects not to approve user
4. funds are now locked with unapproved user

not a super big deal though

Reply:
True -- what do you think we should do with the tiering system
https://github.com/teamsempo/SempoBlockchain/pull/97#discussion_r366329496